pkg/apis/core/types.go

```diff
@@ -4572,7 +4572,6 @@ type Event struct {
 	metav1.ObjectMeta
 
 	// Required. The object that this event is about. Mapped to events.Event.regarding
-	// +optional
```

Comment:
Just to check - is that the only field that effecitvely isn't optional?

Can you please double check if fields like FirstTimestamp isn't effectively required too?
[You can do that by trying to create an Event with just metadata and InvolvedObject being set.]

Reply:
Sure I ll do so.

Reply:
optional comments on the internal API types don't actually drive any behavior… is there a corresponding optional comment on the core/v1 or events/v1 external type we need to remove?

Reply:
@liggitt , under core/v1, it seems pretty much sorted but in events/v1, `ObjectMeta` (metadata) was specified as "+Optional". Hence, I edited it. Apart from that, I correspondingly edited in the internal API as well.

@wojtek-t I checked and confirmed. Under event, only metadata and involvedObject keys are necessary to drive out the successful creation of the `Event` object. So, rest seems fine.